contracts/PricerRule.sol

```diff
@@ -0,0 +1,408 @@
+        bytes3 _payCurrencyCode
+    )
+        external
+        onlyWorker
+    {
+        PriceOracleInterface priceOracle = baseCurrencyPriceOracles[
+            _payCurrencyCode
+        ];
+
+        require(
+            priceOracle != address(0),
+            "Price oracle to remove does not exist."
+        );
+
+        delete baseCurrencyPriceOracles[_payCurrencyCode];
+
+        emit PriceOracleRemoved(priceOracle);
+    }
+
+    /**
+     * @notice Adds an acceptance margin of the base currency price in the
+     *         specified pay currency.
+     *
+     * @dev Function requires:
+     *          - Only organization's workers are allowed to call the function.
+     *          - The specified pay currency code is not null.
+     */
+    function addAcceptanceMargin(
+        bytes3 _payCurrencyCode,
+        uint256 _acceptanceMargin
+    )
+        external
+        onlyWorker
+    {
+        require(
+            _payCurrencyCode != bytes3(0),
+            "Pay currency code is null."
+        );
+
+        baseCurrencyPriceAcceptanceMargins[_payCurrencyCode] = _acceptanceMargin;
+
+        emit AcceptanceMarginAdded(
+            _payCurrencyCode,
+            _acceptanceMargin
+        );
+    }
+
+    /**
+     * @notice Removes an acceptance margin of the base currency price in the
+     *         specified pay currency.
+     *
+     * @dev Function requires:
+     *          - Only organization's workers are allowed to call the function.
+     *          - The specified pay currency code is not null.
+     */
+    function removeAcceptanceMargin(
+        bytes3 _payCurrencyCode
+    )
+        external
+        onlyWorker
+    {
+        require(
+            _payCurrencyCode != bytes3(0),
+            "Pay currency code is null."
+        );
+
+        delete baseCurrencyPriceAcceptanceMargins[_payCurrencyCode];
+
+        emit AcceptanceMarginRemoved(_payCurrencyCode);
+    }
+
+
+    /* Private Functions */
+
+    function getBaseCurrencyPrice(
+        bytes3 _payCurrencyCode
+    )
+        private
+        view
+        returns(uint256)
+    {
+        PriceOracleInterface priceOracle = baseCurrencyPriceOracles[
+            _payCurrencyCode
+        ];
+
+        require(
+            priceOracle != address(0),
+            "Price oracle for the specified currency code does not exist."
+        );
+
+        return priceOracle.getPrice();
+    }
+
+    function convertPayCurrencyToToken(
+        uint256 _baseCurrencyPriceInPayCurrency,
+        uint256 _payCurrencyAmount
+    )
+        private
+        view
+        returns (uint256)
+    {
+        return _payCurrencyAmount
+            .mul(
+                conversionRateFromBaseCurrencyToToken
+            )
+            .div(
+                10 ** conversionRateDecimalsFromBaseCurrencyToToken
+            )
+            .div(
+                _baseCurrencyPriceInPayCurrency
+            );
+    }
+
+    function isPriceInRange(
```

Comment:
Assuming it works / passes tests, 👍 for the refactor!

Reply:
It's not yet, I am currently on it :).